### PR TITLE
Converting formatter into its own package

### PR DESCRIPTION
**Pull request by JordanBoltonMN**
Moves formatter into `@microsoft/powerquery-formatter`.

### CODE REVIEW

File: src/language-services/formatter.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as PQF from "@microsoft/powerquery-formatter";
+import * as PQP from "@microsoft/powerquery-parser";
+
+import { FormattingOptions, Range, TextDocument, TextEdit } from "vscode-languageserver-types";
+
+export function tryFormatconnection(
```
Comment (mattmasson) [inline src/language-services/formatter.ts]: huh?

File: src/language-services/formatter.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    const triedFormat: PQF.TriedFormat = PQF.tryFormat(pqfFormatSettings, document.getText());
+
+    if (PQP.ResultUtils.isOk(triedFormat)) {
+        return PQP.ResultUtils.okFactory([TextEdit.replace(fullDocumentRange(document), triedFormat.value)]);
+    } else {
+        let message: string;
+        if (PQF.FormatError.isTFormatError(triedFormat.error)) {
+            message = triedFormat.error.innerError.message;
+        } else {
+            message = "An unknown error occured during formatting.";
```
Comment (mattmasson) [inline src/language-services/formatter.ts:36]: Do we have other cases where we have to localize strings in language-services?

File: src/language-services/formatter.ts (hunk 1 above)
Comment (JordanBoltonMN) [reply src/language-services/formatter.ts]: Fixed in a previous commit. Closed.

File: src/language-services/formatter.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import * as PQF from "@microsoft/powerquery-formatter";
+import * as PQP from "@microsoft/powerquery-parser";
+
+import { FormattingOptions, Range, TextDocument, TextEdit } from "vscode-languageserver-types";
+
+export function tryFormatconnection(
+    document: TextDocument,
+    formattingOptions: FormattingOptions,
+    localizationTemplates: PQP.ILocalizationTemplates,
+): PQP.Result<TextEdit[], string> {
```
Comment (mattmasson) [inline src/language-services/formatter.ts:13]: Can we return something that isn't parser specific?

Comment (JordanBoltonMN) [reply src/language-services/formatter.ts]: Fixed.

File: src/language-services/formatter.ts (hunk 2 above)
Comment (JordanBoltonMN) [reply src/language-services/formatter.ts]: Discussed in Teams:

We're going to either throw an error message from the parser (already localized) or throw undefined and let VSCode handle it from there.

File: src/test/language-services/inspection.ts [hunk 4]
```diff
@@ -5,15 +5,18 @@ import * as PQP from "@microsoft/powerquery-parser";
...
 import { SignatureProviderContext } from "../../language-services";
 import { InspectionUtils } from "../../language-services";
 import * as Utils from "./utils";
-import { Position } from "vscode-languageserver-types";
 
 // tslint:disable: no-unnecessary-type-assertion
 
 function expectScope(inspected: PQP.Task.InspectionOk, expected: string[]): void {
-    expect(inspected.scope).to.have.keys(expected);
+    const inclusiveScopeKeys: ReadonlyArray<string> = [...inspected.scope.entries()]
```
Comment (JordanBoltonMN) [inline src/test/language-services/inspection.ts:16]: Scope was reworked to invoke recursive scope members. This updates the test to filter those recursive members out.

**Pull request merged**